### PR TITLE
Upgrade pyre

### PR DESCRIPTION
**Pull request by lpetre**
This PR

 - upgrades pyre to 0.9.9 which brings a new type checking backend
- fixes some amount of type errors that 0.9.9 detects now
- suppresses the rest of the type errors

### CODE REVIEW

File: .pyre_configuration [hunk 1]
```diff
@@ -5,5 +5,6 @@
     "exclude": [
         ".*/\\.tox/.*"
     ],
+    "workers": 3,
```
Comment (lisroach) [inline .pyre_configuration:8]: What's the Pyre bug this fixes?

Comment (lpetre) [reply .pyre_configuration:8]: It is an issue where workers < 3 causes pyre to SIGSEGV. We discovered the issue (and the workaround) in this PR https://github.com/Instagram/LibCST/pull/579

Comment (lisroach) [reply .pyre_configuration:8]: Yikes, thanks for the conext!

**Review by lisroach — APPROVED** [review]

File: fixit/cli/__init__.py [hunk 2]
```diff
@@ -215,6 +217,7 @@ def run_ipc(
     Returns an IPCResult object.
     """
 
+    # pyre-fixme[35]: Target cannot be annotated.
     paths: Generator[str, None, None] = (
         os.path.join(prefix, p) if prefix else p for p in paths
     )
```
Comment (zsol) [inline fixit/cli/__init__.py:223]: This should be easy to fix, it's just a name shadowing problem.

File: fixit/cli/apply_fix.py [hunk 3]
```diff
@@ -144,6 +144,9 @@ def apply_fix_operation(
                         get_lint_config().formatter, updated_source
                     )
                 with open(path, "wb") as f:
+                    # pyre-fixme[6]: Expected `Union[array.array[typing.Any],
+                    #  bytearray, bytes, ctypes._CData, memoryview, mmap.mmap]` for 1st
+                    #  param but got `str`.
```
Comment (zsol) [inline fixit/cli/apply_fix.py:149]: Oh cool, this (and the one below) are genuine bugs :) If `updated_source` has some non-ASCII characters, Fixit will clobber the file with invalid encoding.

We should open the file without binary mode:
```suggestion
                with open(path, "w") as f:
```

File: fixit/common/generate_pyre_fixtures.py [hunk 4]
```diff
@@ -59,6 +59,7 @@ def gen_types_for_test_case(source_code: str, dest_path: Path) -> None:
         if "error" in data:
             raise PyreQueryError(cmd, data["error"])
         data = data["response"][0]
+        # pyre-fixme[35]: Target cannot be annotated.
```
Comment (zsol) [inline fixit/common/generate_pyre_fixtures.py:62]: This should be easy to fix too, just rename occurrences of `data` before this line to `raw_data`

**Pull request merged**